spec/meilisearch/index/documents_spec.rb

```diff
@@ -195,31 +196,31 @@
     it 'deletes multiples documents from index' do
       docs_to_delete = [1, 4]
       response = index.delete_documents(docs_to_delete)
-      sleep(0.1)
+      index.wait_for_pending_update(response['updateId'])
       expect(response).to be_a(Hash)
       expect(response).to have_key('updateId')
       expect(index.documents.size).to eq(documents.count - 3 - docs_to_delete.count)
     end
 
     it 'clears all documents from index' do
       response = index.delete_all_documents
-      sleep(0.1)
+      index.wait_for_pending_update(response['updateId'])
       expect(response).to be_a(Hash)
       expect(response).to have_key('updateId')
       expect(index.documents).to be_empty
       expect(index.documents.size).to eq(0)
     end
 
     it 'fails to add document with bad primary-key format' do
-      res = index.add_documents(objectId: 'toto et titi', title: 'Unknown')
-      sleep(0.1)
-      expect(index.get_update_status(res['updateId'])['status']).to eq('failed')
+      response = index.add_documents(objectId: 'toto et titi', title: 'Unknown')
+      index.wait_for_pending_update(response['updateId'])
+      expect(index.get_update_status(response['updateId'])['status']).to eq('failed')
```

Comment:
```suggestion
      expect(index.get_update_status(response['updateId'])['status']).to eq('failed')
      index.wait_for_pending_update(response['updateId'])
```

Change order so that error will come from the `expect` and not from wait_for_... Also consistency with other tests

Reply:
If I change the order, there is no point to wait for the update. Remember, there is no error thrown by the `wait_for_pending_update` method -> it only returns the update.

Reply:
I don't really understand what is not consistent with the other tests:
- we launch the asynchronous task
- we wait for the end of the update with `wait_for_pending_updat`
- we check the status of the update -> in this test we check we get a `failed` status

Reply:
Ah yes my bad I read it as `expect response to have updateId`